pages/register/register.controller.js

```diff
@@ -0,0 +1,164 @@
+(function () {
+	angular.module('app').controller('RegisterController', ['register', 'login', 'session', '$state', '$stateParams', '$translate', RegisterController]);
+
+	function RegisterController(register, login, session, $state, $stateParams, $translate) {
+		var self = this;
+
+		// Service binding
+		self.service = register;
+
+		// Properties
+		self.username = '';
+		self.email = '';
+		self.password = '';
+		self.age = '';
+		self.terms = false;
+		self.error = '';
+		self.loading = false;
+		self.status = {
+			register: true,
+			terms: false,
+			captcha: false,
+		};
+
+		// Methods
+		self.list1 = list1;
+		self.list2 = list2;
+		self.submit = submit;
+
+		// On Run...
+		if (session.id > 0) {
+			console.log('Found a session! Redirecting...');
+			$state.go($stateParams.redirect || 'home');
+		}
+
+		// Internal functions
+		function list1() {
+			self.loading = true;
+			self.error = null;
+			console.log('Verifying...');
+			validateSlide1(function (error, message) {
```

Comment:
Adding Angular Validators (I'm on that), you don't need to validate anything in the controller. This comment also applies to every validation in this PR. But, as I've just said, don't worry, I'm adding validators ;)

Reply:
Hmmm... I was thinking about that, but in this case, we have 2 different slides, both with fields. Get ready to include some ng-if... instead of novalidate on a form & handling the error in the controller